sup3r/bias/bias_calc.py

```diff
@@ -0,0 +1,618 @@
+            sup3r bias calc config with all necessary args and kwargs to
+            initialize the class and call run() on a single node.
+        """
+        import_str = 'import time;\n'
+        import_str += 'from reV.pipeline.status import Status;\n'
+        import_str += 'from rex import init_logger;\n'
+        import_str += f'from sup3r.bias.bias_calc import {cls.__name__};\n'
+
+        if not hasattr(cls, 'run'):
+            msg = ('I can only get you a node command for subclasses of '
+                   'DataRetrievalBase with a run() method.')
+            logger.error(msg)
+            raise NotImplementedError(msg)
+
+        # pylint: disable=E1101
+        init_str = get_fun_call_str(cls, config)
+        fun_str = get_fun_call_str(cls.run, config)
+        fun_str = fun_str.partition('.')[-1]
+        fun_str = 'bc.' + fun_str
+
+        log_file = config.get('log_file', None)
+        log_level = config.get('log_level', 'INFO')
+        log_arg_str = (f'"sup3r", log_level="{log_level}"')
+        if log_file is not None:
+            log_arg_str += f', log_file="{log_file}"'
+
+        cmd = (f"python -c \'{import_str}\n"
+               "t0 = time.time();\n"
+               f"logger = init_logger({log_arg_str});\n"
+               f"bc = {init_str};\n"
+               f"{fun_str};\n"
+               "t_elap = time.time() - t0;\n")
+
+        job_name = config.get('job_name', None)
+        status_dir = config.get('status_dir', None)
+        if job_name is not None and status_dir is not None:
+            status_file_arg_str = f'"{status_dir}", '
+            status_file_arg_str += f'module="{ModuleName.BIAS_CALC}", '
+            status_file_arg_str += f'job_name="{job_name}", '
+            status_file_arg_str += 'attrs=job_attrs'
+
+            cmd += ('job_attrs = {};\n'.format(json.dumps(config)
+                                               .replace("null", "None")
+                                               .replace("false", "False")
+                                               .replace("true", "True")))
+            cmd += 'job_attrs.update({"job_status": "successful"});\n'
+            cmd += 'job_attrs.update({"time": t_elap});\n'
+            cmd += f'Status.make_job_file({status_file_arg_str});\n'
+
+        cmd += "\'"
+
+        return cmd.replace('\\', '/')
+
+    def get_bias_gid(self, coord):
+        """Get the bias gid from a coordinate.
+
+        Parameters
+        ----------
+        coord : tuple
+            (lat, lon) to get data for.
+
+        Returns
+        -------
+        bias_gid : int
+            gid of the data to retrieve in the bias data source raster data.
+            The gids for this data source are the enumerated indices of the
+            flattened coordinate array.
+        d : float
+            Distance in decimal degrees from coord to bias gid
+        """
+        d, i = self.bias_tree.query(coord)
+        bias_gid = self.bias_gid_raster.flatten()[i]
```

Comment:
Include a check that the index is in the raster?

Reply:
there's no distance threshold on this method so `i` should always be in the gid raster. I don't include a distance threshold because this is just a convenience method for debugging and investigation, not used in the actual bc module in a pipeline.

Reply:
There's no default threshold?

Reply:
nope its infinite haha

Reply:
oh lol